intents_consolidator/src/intents/consolidator/consolidator.cpp

```diff
@@ -23,12 +23,23 @@ void Consolidator::Loop()
 {
   while (true) {
     // Pop elem and then run merge on non-null type in result
+    const std::lock_guard<std::mutex> lock{queue_mutex_};
+    if (queue_.empty()) {
+      continue;
```

Comment:
This will busy wait. I think it could potentially be better to create our own syncrhonized queue datastructure (using std::queue internally) with blocking push and pop functions.